h5grove/utils.py

```diff
@@ -125,17 +124,29 @@ def _sanitize_dtype(dtype: np.dtype) -> np.dtype:
     return result
 
 
-def sanitize_array(array: Sequence[Number], copy: bool = True) -> np.ndarray:
-    """Ensure array save as .npy can be read back by js-numpy-parser.
+def convert(
+    data: Union[np.ndarray, np.number, np.bool_], dtype: Optional[str] = "origin"
+) -> Union[np.ndarray, np.number, np.bool_]:
+    """Convert array or numpy scalar to given dtype query param
 
-    See https://github.com/ludwigschubert/js-numpy-parser
-
-    :param array: Array to sanitize
-    :param copy: Set to False to avoid copy if possible
-    :raises ValueError: For unsupported array dtype
+    :param data: nD array or scalar to convert
+    :param dtype: Data type conversion query parameter
+        - `origin` (default): No conversion
+        - `safe`: Convert to a type supported by JS typedarray (https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/TypedArray)
     """
-    ndarray = np.array(array, copy=False)
-    return np.array(ndarray, copy=copy, order="C", dtype=_sanitize_dtype(ndarray.dtype))
+    if dtype in ("origin", None):
+        return data
+
+    if dtype == "safe":
+        if not is_numeric_data(data):
+            raise ValueError(f"Unsupported dtype {dtype} for non-numeric content")
+        return data.astype(_sanitize_dtype(data.dtype), order="C", copy=False)
+
+    raise ValueError(f"Unsupported dtype {dtype}")
+
+
+def is_numeric_data(data: np.ndarray) -> bool:
```

Comment:
```suggestion
def is_numeric_data(data: Union[np.ndarray, np.number, np.bool_]) -> bool:
```

to be in line with `convert`